source/monitoring/cluster-monitoring.txt

```diff
@@ -0,0 +1,249 @@
+  D, [2018-10-09T13:58:03.496504 #22079] DEBUG -- : SDAM | Server description for localhost:27102 changed from 'unknown' to 'secondary'.
+  D, [2018-10-09T13:58:03.496777 #22079] DEBUG -- : SDAM | Topology type 'Unknown' changed to type 'ReplicaSetNoPrimary'.
+  D, [2018-10-09T13:58:03.497306 #22079] DEBUG -- : SDAM | Server 127.0.0.1:27100 connection closed.
+  D, [2018-10-09T13:58:03.497606 #22079] DEBUG -- : SDAM | Topology type 'ReplicaSetNoPrimary' changed to type 'ReplicaSetWithPrimary'.
+
+  # client.close
+
+  D, [2018-10-09T13:58:05.342057 #22079] DEBUG -- : SDAM | Server localhost:27100 connection closed.
+  D, [2018-10-09T13:58:05.342299 #22079] DEBUG -- : SDAM | Server localhost:27101 connection closed.
+  D, [2018-10-09T13:58:05.342565 #22079] DEBUG -- : SDAM | Server localhost:27102 connection closed.
+  D, [2018-10-09T13:58:05.342693 #22079] DEBUG -- : SDAM | Topology type 'ReplicaSetWithPrimary' closed.
+
+.. note::
+
+  The ``:sdam_proc`` client option applies only to the given client. 
+  When certain client options are changed via the
+  ``Client#with`` call, a new cluster may be created by the driver with
+  a default set of event subscribers. If this happens, the provided
+  ``:sdam_proc`` is not called, and the application may miss events.
+
+Server Heartbeats
+~~~~~~~~~~~~~~~~~
+
+You can also create a custom subscriber to monitor server heartbeats.
+
+Custom server heartbeat subscribers differ from other SDAM subscribers,
+as they must implement the following three methods: 
+
+- ``started``: Invoked when the listener receives the heartbeat
+- ``succeeded``: Response for a successful heartbeat outcome
+- ``failed``: Response for a failed heartbeat outcome
+
+The following code is an example heartbeat event subscriber:
+
+.. literalinclude:: /includes/monitoring/monitoring.rb
+   :start-after: start-heartbeat
+   :end-before: end-heartbeat
+   :language: ruby
+   :copyable:
+   :dedent:
+
+You can subscribe to heartbeat events globally or for a specific client:
+
+.. literalinclude:: /includes/monitoring/monitoring.rb
+   :start-after: start-heartbeat-subscribe
+   :end-before: end-heartbeat-subscribe
+   :language: ruby
+   :copyable:
+   :dedent:
+
+When you run the application, your subscriber records the heartbeat event and 
+outputs messages such as the following:
+
+.. code-block:: none
+
+   D, [2018-09-23T13:44:10.707018 #1739] DEBUG -- : HEARTBEAT | 127.0.0.1:27027 | STARTED
+   D, [2018-09-23T13:44:10.707778 #1739] DEBUG -- : HEARTBEAT | 127.0.0.1:27027 | SUCCEEDED | 0.000772381s
+
+.. _SDAM-event-descriptions:
+
+Event Descriptions
+------------------
+
+The following table provides the name and description of each SDAM event:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 40 60
+
+   * - Event Type
+     - Description 
+
+   * - `ServerClosed <{+api-root+}/Mongo/Monitoring/Event/ServerClosed.html>`__
+     - Event fired when the server instance is closed.
```

Comment:
S: to [avoid jargon](https://www.mongodb.com/docs/meta/style-guide/terminology/terms-for-global-audience/#use-jargon-with-care) that might be confusing, you could say "generated" instead of "fired"? or "created"?
```suggestion
     - Event created when the server instance is closed.
```